src/Navigation/Plugins/recastJSPlugin.ts

```diff
@@ -261,9 +275,13 @@ export class RecastJSPlugin implements INavigationEnginePlugin {
      * @returns the resulting point along the navmesh
      */
     moveAlong(position: Vector3, destination: Vector3): Vector3 {
-        var p = new this.bjsRECAST.Vec3(position.x, position.y, position.z);
-        var d = new this.bjsRECAST.Vec3(destination.x, destination.y, destination.z);
-        var ret = this.navMesh.moveAlong(p, d);
+        this._tempVec1.x = position.x;

```

Comment:
Not related to the PR, but I think `moveAlong` should simply rely on `moveAlongToRef` to avoid duplicate code.